src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -708,19 +708,36 @@ class ColumnReferenceSegment(ObjectReferenceSegment):  # type: ignore
     )
 
     def extract_possible_references(self, level):
-        """Extract possible references of a given level."""
+        """Extract possible references of a given level.
+
+        Overrides the parent-class function. BigQuery's support for things like
+        the following:
+        - Functions that take a table as a parameter (e.g. TO_JSON_STRING)
+          https://cloud.google.com/bigquery/docs/reference/standard-sql/
+          json_functions#to_json_string
+        - STRUCT
+
+        means that, without schema information (which SQLFluff does not have),
+        references to data are often ambiguous.
+        """
         level = self._level_to_int(level)
         refs = list(self.iter_raw_references())
         if level == self.ObjectReferenceLevel.SCHEMA.value and len(refs) >= 3:
             return [refs[0]]  # pragma: no cover
-        if level == self.ObjectReferenceLevel.TABLE.value and len(refs) >= 3:
-            # Ambiguous case: The table could be the first or second part, so
-            # return both.
-            return [refs[0], refs[1]]
-        if level == self.ObjectReferenceLevel.OBJECT.value and len(refs) >= 3:
+        if level == self.ObjectReferenceLevel.TABLE.value:
+            # One part: Could be a table, e.g. TO_JSON_STRING(t)
+            # Two parts: Could be dataset.table or table.column.
+            # Three parts: Could be table.column.struct or dataset.table.column.
+            # Four parts: dataset.table.column.struct
+            # Five parts: project.dataset.table.column.struct
+            # So... return the first 3 parts.
+            return refs[:3]
+        if (
+            level == self.ObjectReferenceLevel.OBJECT.value and len(refs) >= 3
+        ):  # pragma: no cover
             # Ambiguous case: The object (i.e. column) could be the first or
             # second part, so return both.
-            return [refs[1], refs[2]]  # pragma: no cover
+            return [refs[1], refs[2]]
         return super().extract_possible_references(level)
```

Comment:
I think this might now need a pragma no cover to fix your coverage issue?

```suggestion
        return super().extract_possible_references(level)  # pragma: no cover
```